src/controllers/sheltersController.js

```diff
@@ -1,47 +1,102 @@
 const sheltersModel = require('../models/sheltersModel.js');
 const { inputValidation } = require("../utils/tools.js");
 
-const { getAllShelters, createShelters } = sheltersModel;
+const { getAllShelters, createShelters, getShelterByID, deleteShelterByID, updateShelterByID } = sheltersModel;
 
 const getShelters = async (req, res, next) => {
   await getAllShelters()
     .then((dbResponse) => {
       res.send(dbResponse);
     })
     .catch((e) => {
-      console.log(e.message);
+      console.log(e);
       res.sendStatus(500);
       next(e);
     });
 };
 
 const postShelters = async (req, res, next) => {
 
-  var attrs = ["address", "emailAddress", "password", "phoneNumber", "shelterName"];
-  var contype = req.headers['content-type'];
+  var success = true;
   //check content type
-  if (!contype || contype.indexOf('application/json') !== 0) res.sendStatus(406);
+  success = inputValidation.checkContentType(req, res);
   //check all attrs are provided (input validation)
-  else if (!inputValidation.hasAllAttrs(req.body, attrs)) res.status(400).send("Please provide required input attributes");
+  var attrs = ["address", "emailAddress", "password", "phoneNumber", "shelterName"];
+  if (success) success = inputValidation.checkAttrs(res, req.body, attrs);
   //check null values
-  else if (inputValidation.includesNullorEmpty(req.body, ["website"])) res.status(400).send("Please provide required input values");
-  else {
+  if (success) success = inputValidation.checkNullorEmpty(res, req.body, ["website"]);
```

Comment:
Are there any exceptions that come up in the API when this is encountered? It seems like `checkAttrs` will res.send an error message, but then this controller will keep moving forward, which will cause an exception down the line. Does this need to be an `else if`? Or am I misunderstanding?